src/views/Swap/index.tsx

```diff
@@ -214,10 +216,35 @@ export default function Swap({ history }: RouteComponentProps) {
     setSwapState({ tradeToConfirm: trade, swapErrorMessage, txHash, attemptingTxn })
   }, [attemptingTxn, swapErrorMessage, trade, txHash])
 
+  // swap warning state
+  const [swapWarningCurrency, setSwapWarningCurrency] = useState(null)
+  const [onPresentSwapWarningModal] = useModal(<SwapWarningModal swapCurrency={swapWarningCurrency} />)
+
+  const shouldShowSwapWarning = (swapCurrency) => {
```

Comment:
Logic to see if selected currency's address matches any address within the `warningTokenConfig`